src/transformers/generation/utils.py

```diff
@@ -3225,7 +3226,10 @@ def beam_search(
                 outputs, model_kwargs, is_encoder_decoder=self.config.is_encoder_decoder
             )
             if model_kwargs["past_key_values"] is not None:
+                using_new_cache = isinstance(model_kwargs["past_key_values"], Cache)
                 model_kwargs["past_key_values"] = self._reorder_cache(model_kwargs["past_key_values"], beam_idx)
+                if using_new_cache:
```

Comment:
```suggestion
                if not isinstance(model_kwargs["past_key_values"], Cache):
```
Shouldn't it be like this?

Reply:
Ah it's because `self._reorder_cache(...)` return a tuple? I'd be quite heavily in favor of adding a `_reorder_cache` to the `Cache` object. Then because in `generate` we only pass around objects of type `Cache` we can delete `_reorder_cache` completely no?